taskcluster/scripts/toolchain/cross-compile-setup.sh

```diff
@@ -25,7 +25,11 @@ tooltool.py \
   --manifest="/builds/worker/checkouts/src/libs/macos-cc-tools.manifest" \
   fetch
 
+popd || exit
+
+# It's important that we execute rustup in the checkout directory, so that it can see
+# any `rust-toolchain` override file that might be present.
```

Comment:
This, it turns out, was the source of the problem with the release builds. I don't really understand *why*, because AFAICT we were setting the rustup default toolchain to the same toolchain that's specified in our `rust-toolchain` file. But anyway, running this from the repo root rather than from `/tmp` seems to have fixed the issue.